src/Storages/MergeTree/MergedBlockOutputStream.cpp

```diff
@@ -211,6 +212,8 @@ MergedBlockOutputStream::WrittenFiles MergedBlockOutputStream::finalizePartOnDis
             auto count_out = new_part->getDataPartStorage().writeFile("count.txt", 4096, write_settings);
             HashingWriteBuffer count_out_hashing(*count_out);
             writeIntText(rows_count, count_out_hashing);
+            count_out_hashing.write(",", 1);
+            writeIntText(masked_rows_count, count_out_hashing);
```

Comment:
TODO: Writing this metadata into a different file might be better for forwards compatibility.